includes/api/class-wc-rest-shipping-zones-controller.php

```diff
@@ -0,0 +1,246 @@
+	 */
+	public function register_routes() {
+		register_rest_route( $this->namespace, '/' . $this->rest_base, array(
+			array(
+				'methods'             => WP_REST_Server::READABLE,
+				'callback'            => array( $this, 'get_items' ),
+				'permission_callback' => array( $this, 'get_items_permissions_check' ),
+			),
+			array(
+				'methods'             => WP_REST_Server::CREATABLE,
+				'callback'            => array( $this, 'create_item' ),
+				'permission_callback' => array( $this, 'update_items_permissions_check' ),
+				'args'                => $this->get_endpoint_args_for_item_schema( WP_REST_Server::CREATABLE ),
+			),
+			'schema' => array( $this, 'get_public_item_schema' ),
+		) );
+
+		register_rest_route( $this->namespace, '/' . $this->rest_base . '/(?P<id>[\d-]+)', array(
+			array(
+				'methods'             => WP_REST_Server::READABLE,
+				'callback'            => array( $this, 'get_item' ),
+				'permission_callback' => array( $this, 'get_items_permissions_check' ),
+			),
+			array(
+				'methods'             => WP_REST_Server::EDITABLE,
+				'callback'            => array( $this, 'update_item' ),
+				'permission_callback' => array( $this, 'update_items_permissions_check' ),
+				'args'                => $this->get_endpoint_args_for_item_schema( WP_REST_Server::EDITABLE ),
+			),
+			'schema' => array( $this, 'get_public_item_schema' ),
+		) );
+	}
+
+	/**
+	 * Get a single Shipping Zone.
+	 *
+	 * @param WP_REST_Request $request
+	 * @return WP_REST_Response
+	 */
+	public function get_item( $request ) {
+		$zone = $this->get_zone( $request->get_param( 'id' ) );
+
+		if ( is_wp_error( $zone ) ) {
+			return $zone;
+		}
+
+		$data = $zone->get_data();
+		$data = $this->prepare_item_for_response( $data, $request );
+		$data = $this->prepare_response_for_collection( $data );
+
+		return rest_ensure_response( $data );
+	}
+
+	/**
+	 * Get all Shipping Zones.
+	 *
+	 * @param WP_REST_Request $request
+	 * @return WP_REST_Response
+	 */
+	public function get_items( $request ) {
+		$rest_of_the_world = WC_Shipping_Zones::get_zone_by( 'zone_id', 0 );
+
+		$zones = WC_Shipping_Zones::get_zones();
+		array_unshift( $zones, $rest_of_the_world->get_data() );
+		$data  = array();
+
+		foreach ( $zones as $zone_obj ) {
+			$zone   = $this->prepare_item_for_response( $zone_obj, $request );
+			$zone   = $this->prepare_response_for_collection( $zone );
+			$data[] = $zone;
+		}
+
+		return rest_ensure_response( $data );
+	}
+
+	/**
+	 * Create a single Shipping Zone.
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return WP_REST_Request|WP_Error
+	 */
+	public function create_item( $request ) {
+		$zone = new WC_Shipping_Zone( null );
+
+		if ( ! is_null( $request->get_param( 'name' ) ) ) {
+			$zone->set_zone_name( $request->get_param( 'name' ) );
+		}
+
+		if ( ! is_null( $request->get_param( 'order' ) ) ) {
+			$zone->set_zone_order( $request->get_param( 'order' ) );
+		}
+
+		$zone->create();
```

Comment:
We should have some error checking here to make sure the zone is actually created correctly. `$zone->get_id()` should !== 0.